### ISSUE TITLE
Weird issue with aria and hasAttribute

### ISSUE THREAD

**Issue by MarcosPortugal** [angular/material#1282]
I have run into a weird issue, where when I drop an object on a canvas (lots of stuff going on, can't really tell what the problem really is) I get the error "Uncaught TypeError: Cannot read property 'hasAttribute' of undefined".

 The lines causing this error, from the angular material js file, are:

```javascript
 /**
   * Check if expected attribute has been specified on the target element or child
   * @param element
   * @param attrName
   * @param {optional} defaultValue What to set the attr to if no value is found
   */
  function expect(element, attrName, defaultValue) {
    var node = element[0];

    /* this line right here --> */ if (!node.hasAttribute(attrName) && !childHasAttribute(node, attrName)) {

      defaultValue = angular.isString(defaultValue) && defaultValue.trim() || '';
      if (defaultValue.length) {
        element.attr(attrName, defaultValue);
      } else {
        $log.warn('ARIA: Attribute "', attrName, '", required for accessibility, is missing on node:', node);
      }

    }
  }
```

 Since I really can't tell what is causing this error, and it seems it would be a tedious task to dig around for a cause, I would just like to know how to disable the aria functionality included in angular material.

 I am using the files from the 0.7.0 release.

**Comment by marcysutton** [comment]
I would want to debug the issue before disabling ARIA functionality. What directives are you using? It must be an element with an expect() on it. If you can post something in a Codepen or Plunker, I can help you debug.

**Comment by MarcosPortugal** [comment]
Well, this morning I was throwing around some console.logs to see what was going on right before the error, and to my surprise it's not happening anymore. Pretty weird. I'll keep an eye open, but this will be closed for now. Thank you very much for your time! :)

**Comment by birdcw** [comment]
This should not be closed. Any md-button without an aria-label throws the same uncaught error for me. angular-material.js line 824. It can take a long time to search through the html to find the offending md-button with no arial-label. A huge waste of time of those of use who could care less about accessibility. This uncaught error makes angular-material very unaccessible for me right now indeed.

**Comment by marcysutton** [comment]
@birdcw can you provide a Plunker or Codepen showing your issue? I am looking at our demos for buttons and they do not throw errors. You only need `aria-label` if it's an icon button, meaning it doesn't have text content. If there is no text content or label, it will print the offending node to the console so you can find it easily.

**Comment by birdcw** [comment]
Thanks for the quick reply, Marcy. I've just been trying to isolate my problem better in plunker, and you're definitely right that md-buttons do in fact log the node. The problem is when the node is undefined. I haven't been able to pinpoint yet, why nodes are ever undefined for me, it's definitely not the case if I just place an md-button in the index.html, but I have seen it crop up a few times.

I have been able to get the error through a (possibly incorrect) use of $mdDialog with a templateUrl. If my dialog setup is correct, then I think we can start here. What is suspicious to me is that the aria service would through an uncaught error in a situation where no aria attributes should be required, correct?

http://plnkr.co/edit/hpjgYEms2hkHfdc6mqhB

**Comment by birdcw** [comment]
@marcysutton I think the problems I've been having are likely just related to templateUrl. And it might actually just be plunker's fault. I've noticed a lot of 404 errors in plunker recently (an issue with plunker, not the code). I've just switched from templateURL to template, which I'm more than happy to use instead (md is coming along great!). So I haven't tested templateURL outside of plunker.  

It's plausible that something in the aria service is causing the error, and even if it was something else, hopefully there can be a fix to prevent an uncaught exception in the aria service.

I mentioned buttons in my first comment, but the only issue with aria and md-button that I'm sure of is that the console requests an aria-label using the attribute ng-bind in place of double curly braces (and no hard coded text). It would be great if aria-label warnings can behave the same whether using ng-bind or double curlies.

Cheers!
